### ISSUE TITLE
exporting, subsetting, and matching output is quite slow

### ISSUE THREAD

**Issue by jordansread** [USGS-R/glmtools#262]
We haven't taken a look at this code in a _long_ time, but when doing multilake calibrations or different types of calibrations, the `compare_to_field` workflow is pretty slow. This is probably a low priority for the time being, but probably some easy wins using more modern packages to do this type of matching/interpolating within the output matrix. 

Before starting on this it would be good to benchmark with an example.

**Comment by hdugan** [comment]
Yes, good idea. I think the time sink is in `resample_to_field` so I'll start there

**Comment by hdugan** [comment]
@jread-usgs rewrote `resample_to_field` using `join` function from `dplyr`. Improves speed about 25%. The rest of the time sink is in `get_var`

Also, I was trying to figure out how to update my branch with the new commits on the USGS ggoverhaul fork... and I'm not sure how, but managed to commit directly to it, instead of to mine 🤦‍♂. Since I've already messed up, going to leave it as is and not damage anything further. 

Anyway, only changes are in `resample_to_field`
